core/ir/x86/decode_table.c

```diff
@@ -9251,6 +9273,26 @@ const instr_info_t evex_Wb_extensions[][4] = {
     {OP_vsqrtss, 0xf30f5110, "vsqrtss", Vdq, xx, KE1b, Hdq, Uss, mrm|evex|er|ttt1s, x, END_LIST},
     {OP_vsqrtsd, 0xf20f5140, "vsqrtsd", Vdq, xx, KE1b, Hdq, Wsd, mrm|evex|ttt1s, x, tevexwb[266][3]},
     {OP_vsqrtsd, 0xf20f5150, "vsqrtsd", Vdq, xx, KE1b, Hdq, Usd, mrm|evex|er|ttt1s, x, END_LIST},
+  }, { /* evex_W_ext 267 */
+    {OP_vpdpbusd, 0x66385008, "vpdpbusd", Ve, xx, KEd, He, We, mrm|evex|ttfvm, x, END_LIST},
```

Comment:
OK, let's see if I can get the details right here:

The manual says:
```
VEX.128.66.0F38.W0 50 /r     VPDPBUSD xmm1, xmm2, xmm3/m128
VEX.256.66.0F38.W0 50 /r     VPDPBUSD ymm1, ymm2, ymm3/m256
```

VEX means it can be encoded with either vex or evex IIRC?  So this entry should not have `|evex` but rather `|vex`?

"W0" means evex.W has to be 0.  The W=1 entries here are all invalid so that makes sense.

Since it has both 128 and 256, evex.L will decide which (1==256).
So the 256 versions should all have 0x80 set in the bottom byte to indicate the L must be set?
No, I think we're relying on the operands to set evex.L (and the reverse for decoding) for these cases: those aren't different table entries.

There is no broadcast version so no evex.B.

The 0F38 should set OPCODE_THREEBYTES so bottom bits 0x8.

No disp compression and no broadcast: I think that's DR_TUPLE_TYPE_FVM?  I wish there were comments reminding us what these confusing DR tuple type enum names mean (if you want to add that sounds good to me).

The operands:
The manual says it has 3 operands:
`ModRM:reg (r, w) VEX.vvvv (r) ModRM:r/m (r`
This table entry has 4:
```
Ve TYPE_V, OPSZ_16_vex32_evex64
KEd TYPE_K_EVEX, OPSZ_4
He TYPE_H, OPSZ_16_vex32_evex64
We TYPE_W, OPSZ_16_vex32_evex64
```
V is modrm:reg, H is vex.vvvv, W is modrm:r/m.
The K would be a writemask to filter the elements -- but I don't see a mention of masking applying to these instructions?

Reply:
> VEX means it can be encoded with either vex or evex IIRC? So this entry should not have |evex but rather |vex?

I think in your manual listing the evex entries are missing so I think the evex_Wb_extensions should have `|evex` . I changed the vex entries to use `|vex`.
```
AVX-VNNI
VEX.128.66.0F38.W0 50 /r VPDPBUSD xmm1, xmm2, xmm3/m128
VEX.256.66.0F38.W0 50 /r VPDPBUSD ymm1, ymm2, ymm3/m256

AVX512_VNNI
EVEX.128.66.0F38.W0 50 /r VPDPBUSD xmm1{k1}{z}, xmm2, xmm3/m128/m32bcst
EVEX.256.66.0F38.W0 50 /r VPDPBUSD ymm1{k1}{z}, ymm2, ymm3/m256/m32bcst
EVEX.512.66.0F38.W0 50 /r VPDPBUSD zmm1{k1}{z}, zmm2, zmm3/m512/m32bcst
```
> No disp compression and no broadcast: I think that's DR_TUPLE_TYPE_FVM? 

The evex entries do have broadcast so I think I've got this wrong. I don't fully understand this stuff so could use some help with what the tuple type here should be.

>  The K would be a writemask to filter the elements -- but I don't see a mention of masking applying to these instructions?

The evex entries do have a mask so I hope I got this right.

Reply:
> I think in your manual listing the evex entries are missing

Hmm you are correct: I was looking at a manual containing just recent extensions, including AVX-VNNI and including the CPUID bit for AVX512-VNNI but somehow not including actual AVX512-VNNI encodings.  With a recent complete manual I see the evex encodings.

Reply:
> > No disp compression and no broadcast: I think that's DR_TUPLE_TYPE_FVM?
>
> The evex entries do have broadcast so I think I've got this wrong. I don't fully understand this stuff so could use some help with what the tuple type here should be.

I am also not very familiar, we'll have to look at the tables (it was @hgreving2304 who added the AVX-512 support but has moved to other projects).

Reply:
I added broadcast entries to the EVEX_Wb_EXT and changed the tupletype to DR_TUPLE_TYPE_FV to get the binutils tests working.

Reply:
OK, it does seem like FV is the right type.

Reply:
Anything else need before we resolve this conversation?

Reply:
I think we're good.  Was waiting for an official re-review request.